### PR TITLE
Add dropout to BiDAF and the character-level encoder

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: experiment_config/bidaf.json [hunk 1]
```diff
@@ -77,13 +82,12 @@
...
-    "type": "adadelta",
-    "lr": 0.5,
-    "rho": 0.95
+    "type": "sgd",
+    "lr": 0.00001
   },
   "num_epochs": 12,
-  "grad_norm": 100.0,
-  "grad_clipping": 100,
+  "grad_norm": 0.1,
```
Comment (DeNeutoy) [inline experiment_config/bidaf.json]: This seems excessively small?

Comment (matt-gardner) [inline experiment_config/bidaf.json]: I was just playing around with a lot of different values, trying to get something to actually train.  I just removed it entirely, just using gradient clipping, and not normalization, and loss is finally going down.  I think there are too many parameters to have this norm make sense.
